scipy/stats/_binned_statistic.py

```diff
@@ -650,6 +633,41 @@ def binned_statistic_dd(sample, values, statistic='mean',
     return BinnedStatisticddResult(result, edges, binnumbers)
 
 
+def _calc_binned_statistic(Vdim, binnumbers, result, values, stat_func):
+    unique_bin_numbers = np.unique(binnumbers)
+    for vv in builtins.range(Vdim):
+        bin_map = _create_bin_map(binnumbers, unique_bin_numbers, values, vv)
+        for i in unique_bin_numbers:
+            result[vv, i] = stat_func(bin_map[i])
+
+
+def _calc_binned_std(Vdim, binnumbers, result, values):
```

Comment:
why is it needed to have a ***separate*** function for the `_calc_binned_std`? is this purely b/c of the special logic with the bins have <2 entries? If so what is the performance reduction by not checking for <2 entries in bin? I'd guess that the repeated searching/binning is/was the larger performance reduction cause. 

Thoughts?

Reply:
I took a benchmark, there is no big difference performances, so I merged these functions. PTAL